### DISCUSSION TITLE
understanding metplus and other met products

### DISCUSSION THREAD

**Discussion by marstonsward** [dtcenter/METplus#1186] [Configuration] [answered]
Hi,

I installed the metplus docker image and initial tests using standard use cases have worked well. I have, however, some questions:

1.) What filetypes can metplus read?
2.) How do I get plots and configure particular statistics what metplus docker?

I'm looking for an overview of the product and how it can be extended.

**Comment by JohnHalleyGotway** [comment, marked as answer]
Hello Marston,

I see that you've been able to use the METpuls Docker image to run some initial tests. Glad to hear that you're up and running.

In regards to "What filetypes can metplus read?", please read through this section of the [MET User's Guide](https://met.readthedocs.io/en/latest/Users_Guide/data_io.html#input-data-formats). The METplus python wrappers make calls to the MET command line verification tools. And the file types that MET can handle are described in its User's Guide. Generally speaking, it reads gridded data in GRIB1, GRIB2, or a few flavors of NetCDF.

In regards to "How do I get plots and configure particular statistics what metplus docker?", since you're already using Docker, I'd recommend running the METviewer software via Docker as well. To that end, here's a [METplus-Training video](https://metplus-training.readthedocs.io/en/latest/modules/METviewer/docker.html) I created a while back with instructions for running the METviewer software via Docker.

In that video, we just have you load some canned sample data. Once you get METplus creating output, I'd recommend saving it to your machine and then load that data into the METviewer instance. Then you can interactively create plots of it. But I'm guessing you may have lots of questions along the way!

**Reply by marstonsward** [reply]
Hi John,

Thanks for helping me get a bit farther on my way to learning to leverage Met tools. 
I, unfortunately, ran into a problem with docker compose file:

`db:
    image: mysql:5.7
    container_name: mysql_mv
    user: "mysql:mysql"
    ports:
        - "6603:3306"
    volumes:
        - ${MYSQL_DIR}:/mysql

    environment:
        MYSQL_ROOT_PASSWORD: mvuser
    tty: true
metviewer:
    image: ${METVIEWER_DOCKER_IMAGE}
    container_name: metviewer_1
    ports:
    - "8080:8080"
    volumes:
      - ${METVIEWER_DATA}:/metviewer_data
      - ${METVIEWER_DIR}/metviewer_output/xml:/opt/tomcat/webapps/metviewer_output/xml
      - ${METVIEWER_DIR}/metviewer_output/plots:/opt/tomcat/webapps/metviewer_output/plots
      - ${METVIEWER_DIR}/metviewer_output/data:/opt/tomcat/webapps/metviewer_output/data
      - ${METVIEWER_DIR}/metviewer_output/scripts:/opt/tomcat/webapps/metviewer_output/scripts
    links:
      - db:mysql_mv
    tty: true`

I created a `.env`  in the same dir and in this .env file I have defined the env variables:

`MYSQL_DIR=/my_path/metplus/mysql
METVIEWER_DIR=/my_path/metplus/metviewer_output
METVIEWER_DATA=/my_path/metplus/met_out
METVIEWER_DIR=/my_path/metplus/metviewer
METVIEWER_DOCKER_IMAGE=dtcenter/metviewer`

When I run the command:
$ docker-compose --env-file .env up
I get the following, which indicates that there is something wrong with the docker-compose file.
$ (root) Additional property db is not allowed

I've done a cursory examination of the docker-compose file and found that if it is rewritten like this:

`services:
    metviewer:
        image: ${METVIEWER_DOCKER_IMAGE}
        container_name: metviewer_1
        ports:
            - "8080:8080"
        volumes:
            - ${METVIEWER_DATA}:/metviewer_data
            - ${METVIEWER_DIR}/metviewer_output/xml:/opt/tomcat/webapps/metviewer_output/xml
            - ${METVIEWER_DIR}/metviewer_output/plots:/opt/tomcat/webapps/metviewer_output/plots
            - ${METVIEWER_DIR}/metviewer_output/data:/opt/tomcat/webapps/metviewer_output/data
            - ${METVIEWER_DIR}/metviewer_output/scripts:/opt/tomcat/webapps/metviewer_output/scripts
        links:
            - db:mysql_mv
        tty: true
    db:
        image: mysql:5.7
        container_name: mysql_mv
        user: "mysql:mysql"
        ports:
            - "6603:3306"
        volumes:
            - ${MYSQL_DIR}:/mysql
        environment:
            MYSQL_ROOT_PASSWORD: mvuser
        tty: true`

 Then it works. Testing further...

**Reply by marstonsward** [reply]
Thanks John, 

The video and documentation helped. I'm interested in plotting the canned data. Where do I learn how to do that part?
Next step will be to load my own database but I would like to learn to leverage the full MET tool stack first.

**Reply by JohnHalleyGotway** [reply]
Marston, that video includes loading canned MET output data:
https://metplus-training.readthedocs.io/en/latest/modules/METviewer/docker.html
That's about 9:15 into the video.

**Reply by marstonsward** [reply]
Hi John,

I've got it all done. I just need to plot it now. Is there a tutorial on how to use the viewer GUI?

**Reply by marstonsward** [reply]
Hi John,

Hmm...ok. But what do I do in the mean time? Is it possible to setup a lightning walk-through via google meet?

**Comment by JohnHalleyGotway** [comment]
I'd recommend working through the instructions on this page:
https://dtcenter.org/metplus-practical-session-guide-version-4-0/session-5-trkintfeature-relative/metviewer

They use a publicly available version of METviewer rather than the one actually running on your local machine. It's a couple of examples of making plots using a different dataset, but it'll give you the idea.

**Reply by marstonsward** [reply]
I'm not sure I follow the train of thought on how to plot with metviewer. I'll look at it some more. Is there a youtube video of metviewer in action?

**Reply by TaraJensen** [reply]
Marston,  Here is the documentation for METviewer: https://metviewer.readthedocs.io/en/latest/Users_Guide/

In lieu of a YouTube video - which is not ready yet - here's a quick-start guide in Powerpoint form
[METviewer_training_quickstart_20211006.pptx](https://github.com/dtcenter/METplus/files/7295130/METviewer_training_quickstart_20211006.pptx)

**Comment by JohnHalleyGotway** [comment]
Marston,

We do have plans to create those sort of videos to accompany the METplus
online tutorial, as described in these GitHub issues, but they do not yet
exist:

https://github.com/dtcenter/METplus-Training/issues?q=is%3Aopen+is%3Aissue+label%3A%22component%3A+METviewer+Training%22

John

**Reply by JohnHalleyGotway** [reply]
Marston, I'd like to mark this discussion as being answered and "lock it" to prevent future posts. That's how we encourage users to create new discussions for new questions. But I wanted to give you an opportunity to comment on it before doing so. Please feel free to select one of the responses as being the best answer to your original question. That'll help future users with similar questions find answers.

**Reply by marstonsward** [reply]
John,

Sure. Go ahead and close this. It will take some time for me to comb through the documents and learn Metplus.
